conan/tools/meson/meson.py

```diff
@@ -37,6 +37,9 @@ def configure(self, reconfigure=False):
                                                  default=[], check_type=list)
         cmd = "meson setup "
         if is_cross_build:
+            #Do not let pkgconf prefix sysroot path in front of paths to Conan packages (that are 
+            #outside the toolchain sysroot).
+            os.environ['PKG_CONFIG_FDO_SYSROOT_RULES'] = "1"
```

Comment:
Setting an environment variable this way is not correct, it will leave the environment dirty for other packages, etc. If anything, the approach would be in the ``MesonToolchain`` generator, to generate a ``Environment`` script with the necessary env-vars